### ISSUE TITLE
[no-unused-vars] False positive with parameter decorators with default values

### ISSUE THREAD

**Issue by robin-pham** [typescript-eslint/typescript-eslint#1649]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "vars": "all",
        "args": "none"
      }
    ]
  }
}
```

```TS
const log = function(target: Object, method: string, index: number): void {
  console.log('decorating %s[%d]', method, index);
};

class Test {
  public sum(a: number, @log b: number = 7): number {
    return a + b;
  }
}
```

**Expected Result**

No violation of `unused-vars` is reported.

**Actual Result**

It complains that `log` is assigned a value but never used.

**Additional Info**

Note: the rule works fine if there is no default value.

**Versions**
Tried on two sets of versions:

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.21.0` |
| `@typescript-eslint/parser`        | `2.21.0` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.4` |

and

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.1` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
please try switching to our `no-unused-vars-experimental` rule, it will work in this case.

**Comment by robin-pham** [comment]
gotcha, tried it out and it works! 

sadly though our codebase uses options in `no-unused-vars` that don't exist in `no-unused-vars-experimental` :(. 

thanks for the quick reply!

**Comment by bradzacher** [comment]
Options like what?
Potentially could be built into the new rule.

**Comment by bradzacher** [comment]
Merging into #1856
